### ISSUE TITLE
Permission denied?

### ISSUE THREAD

**Issue by brosef** [rlane/rbvmomi#19]
When I run this:

`require 'rbvmomi'
vim = RbVmomi::VIM.connect host: 'vmhost', user: 'miles', password: 'password', insecure: 'true'
dc = vim.serviceInstance.find_datacenter("Labs") or fail "datacenter not found"
vm = dc.find_vm("foo") or fail "VM not found"`

I get:

`/home/miles/.rvm/gems/ruby-1.9.2-p318/gems/rbvmomi-1.2.3/lib/rbvmomi/connection.rb:47:in `parse_response': NoPermission: Permission to perform this operation was denied. (RbVmomi::Fault)
    from /home/miles/.rvm/gems/ruby-1.9.2-p318/gems/rbvmomi-1.2.3/lib/rbvmomi/connection.rb:71:in `call'
    from /home/miles/.rvm/gems/ruby-1.9.2-p318/gems/rbvmomi-1.2.3/lib/rbvmomi/basic_types.rb:182:in `_call'
    from /home/miles/.rvm/gems/ruby-1.9.2-p318/gems/rbvmomi-1.2.3/lib/rbvmomi/basic_types.rb:70:in `block (2 levels) in init'
    from /home/miles/.rvm/gems/ruby-1.9.2-p318/gems/rbvmomi-1.2.3/lib/rbvmomi/vim/Folder.rb:7:in `find'
    from /home/miles/.rvm/gems/ruby-1.9.2-p318/gems/rbvmomi-1.2.3/lib/rbvmomi/vim/Folder.rb:83:in `traverse'
    from /home/miles/.rvm/gems/ruby-1.9.2-p318/gems/rbvmomi-1.2.3/lib/rbvmomi/vim/ServiceInstance.rb:8:in `find_datacenter'
    from RbvmomiTest.rb:3:in `<main>'`

I am a part of the administrator role on this VCenter server, so I should have enough permissions. Any idea what I should be looking at to solve this?

**Comment by skyairmj** [comment]
@brosef, we encountered the same issue today. After digging into the code, we seemed figure out the cause.

The truth is 'serviceInstance.find_datacenter' needs some additional permissions, since it actually invokes queries against the SearchIndex object(which belongs to the ServiceContent object, then eventually belongs to the ServiceInstance object---the root of the vCenter inventory). If the user doesn't have the corresponding permissions to do the query, NoPermission error then is thrown. 

However, if you use 'rootFolder.childEntity.grep(RbVmomi::VIM::Datacenter).find { |x| x.name == "mydatacenter" }' instead, everything goes perfectly. It seems administrator roles are allowed to collect children objects of the root Folder.

So, in conclusion, I'd suggest either you use the latter way to get the datacenter target, or assign the additional permissions to your user. HTH.

**Comment by heathsnow** [comment]
For clarity I'll leave this here (workaround):

``` ruby
# Power Off VM Example
vim = RbVmomi::VIM.connect host: 'MyHost', user: 'domain\myuser', password: 'password', insecure: true

rootFolder = vim.serviceInstance.content.rootFolder

dc = rootFolder.childEntity.grep(RbVmomi::VIM::Datacenter).find { |x| x.name == "MyDC" }

vm = vim.find_vm('Folder/Subfolder/MyComputer') or fail 'VM not found'

vm.PowerOffVM_Task.wait_for_completion
```

**Comment by mrmarbury** [comment]
Do you know by any chance, what the corresponding permission is? I am having this issue with testkitchen, which uses rbvmomi
